### PR TITLE
Fix to version number and move to pyproject.toml usage

### LINKED ISSUE #38: Version number does not seem to be working correctly
Calling `hermes_core.__version__` returns unknown when it should return `0.1.dev`. The docs also show an unknown version number.

### LINKED ISSUE #39: Remove use of towncrier
This is a tool provided by sunpy that allows for the automatic generation of changelogs based on individual issue changes. This requires we provide a description of each merge request and issue closed. This is too bulky. We should just maintain the changelog by hand and ensure that large enough merge requests provide a few sentences describing themselves in the merge request.

### PR DESCRIPTION
**Pull request by ehsteve**
This also removes the use of towncrier for the changelog.

This is a much larger merge request than I anticipated. Once I fixed the version number, I had to update the pyproject.toml file for a dynamic version. I then proceeded to move everything that I could into that file from setup.cfg.

I could not figure out how to set a stable version number, for example, for a release. This is future work.

Closes #39 and closes #38.

### CODE REVIEW

**Comment by ehsteve** [comment]
@dbarrous this needs some significant local testing. Could you try install the package on your local machine with the various optional dependencies please?

File: hermes_core/__init__.py [hunk 1]
```diff
@@ -24,4 +24,4 @@
 INST_TO_SHORTNAME = dict(zip(INST_NAMES, INST_SHORTNAMES))
 INST_TO_TARGETNAME = dict(zip(INST_NAMES, INST_TARGETNAMES))
 
-log.info(f"hermes_core version: {__version__}")
+# log.info(f"hermes_core version: {__version__}")
```
Comment (dbarrous) [inline hermes_core/__init__.py:27]: I was noticed this when testing:
top-before-bottom-after
Was the comment out on purpose?

File: pyproject.toml [hunk 2]
```diff
@@ -1,48 +1,97 @@
 [build-system]
+requires = ["setuptools>=45",
+            "setuptools_scm[toml]>=6.2",
+]
+build-backend = 'setuptools.build_meta'
 
-requires = ["setuptools",
-            "setuptools_scm",
-            "wheel"]
+[project]
```
Comment (dbarrous) [inline pyproject.toml:7]: Verified these changes with pip and this is the Before and After the PR:

**Before** 
Screenshot from 2022-12-09 12-02-45

**After**
pip_show_hermes_core

Should we include the Home-page and Author? It looks like it doesn't parse the name out automatically when defining `authors`

File: pyproject.toml [hunk 3]
```diff
@@ -1,48 +1,97 @@
...
+]
+
+[tool.setuptools.package-data]
+hermes_core = ["data/*"]
+
+[tool.setuptools]
+packages = ["hermes_core"]
+
+[tool.setuptools_scm]
+write_to = "hermes_core/_version.py"
```
Comment (dbarrous) [inline pyproject.toml:69]: I verified this works by copying the package locally from this PR branch and running the `python -m setuptools_scm` command:
setuptool_scm

**Review by dbarrous — APPROVED** [review]
This PR looks good, I left a few comments with some questions but nothing breaking.

**Comment by ehsteve** [comment]
There may be error in this PR. Do not merge.

**Pull request merged**